pkgs/by-name/gt/gtk-session-lock/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, gobject-introspection
+, vala
+, gtk3
+, wayland-scanner
+, wayland
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gtk-session-lock";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "Cu3PO42";
+    repo = "gtk-session-lock";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-SHKAYmdev08oRB/V6UpfSFqYwplF59IaNSOoWcACPig=";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gobject-introspection
+    vala
+  ];
+
+  buildInputs = [
+    gtk3
+    wayland-scanner
+    wayland
+  ];
+
+  # Tests are not fully prepared, but may be enabled in later versions
+  doCheck = false;
+
+  strictDeps = true;
+
+  meta = {
+    description = "A library to use GTK 3 to build screen lockers using ext-session-lock-v1 protocol";
+    homepage = "https://github.com/Cu3PO42/gtk-session-lock";
+    # The author stated "GTK Session Lock is licensed under the GNU General
```

Comment:
That comment means this license is gpl3Plus.

Reply:
>  This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. 

It's "at your option", not "at the author's option", so obviously it's not gpl3Plus. Also see discussion above with the author.

Reply:
> the combined product is available only as GPLv3 or potentially later versions.

This means it is gpl3Plus, gpl3Only would mention that it is not available at any later version.

You are referencing the license text which means that the end user could choose gpl4 if it existed.

Reply:
Upstream even interpreted it this way in their nix file https://github.com/Cu3PO42/gtk-session-lock/blob/b9ddb2792b613d14622acada73c64f16a2635b40/gtk-session-lock.nix#L44

Reply:
The author has explained here https://github.com/Cu3PO42/gtk-session-lock/issues/6#issuecomment-2067202089

> I did intentionally go with the non-plus version there since I'm not pre-approving a potential v4.

The license in that nix file is more likely a misunderstanding.

CC @Cu3PO42

Reply:
I was on the fence on what to put in the `meta` attribute. Neither GPL-3.0-only nor GPL-3.0-or-later fully capture what I want, specifically in regards to contributions. I do agree, however, that GPL-3.0-only is the more reasonable choice and will be changing it upstream.

For now, I am releasing the library as GPLv3 only. Should a later version of GPL be released that I find reasonable for the project, I'd also like to license it as such. However, I also don't want to ask any and all contributors to assign me their copyright. Hence I put the provision "GPL 3 or any later version once I approve it" in the original license that contributors submit patches under.

Seeing all of the confusion this has caused, I will try and think of a different approach.